priv/frontend/src/components/Login/index.js

```diff
@@ -1,9 +1,51 @@
 import React from 'react'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import { Form, Field } from 'simple-react-form'
+import { PasswordField, EmailField } from '../FormComponents/TextFields'
+import * as actions from '../../actions/sessionActions'
 
-const Login = () => (
-    <div className="container">
-        <h1>Login</h1>
-    </div>
-)
+class Login extends React.Component {
+    onSubmit = ({email, password}) => {
+        this.props.dispatch(actions.login(email, password))
+    }
 
-export default Login
+    state = {}
+
+    render() {
+        if (this.props.loading) {
+            return (
+                <div className="container">
+                    <h1>Login</h1>
+                    <p> loading ... </p>
+                </div>
+            )
+        } else if (this.props.error) {
+            return (
+                <div className="container">
+                    <h1>Login</h1>
+                    <p>Error: {this.props.error.toString()} </p>
+                    <Form ref='form' state={this.props.initialDoc} onSubmit={this.onSubmit}>
+						<Field fieldName='email' label='Email' type={EmailField} />
+						<Field fieldName='password' label='Password' type={PasswordField} />
+                    </Form>
+                </div>
+            )
+        } else {
+            return (
+                <div className="container">
+                    <h1>Login</h1>
+                    <Form ref='form' state={this.props.initialDoc} onSubmit={this.onSubmit}>
+                        <Field fieldName='email' label='Email' type={EmailField} />
+                        <Field fieldName='password' label='Password' type={PasswordField} />
+                    </Form>
+                    <button onClick={() => this.refs.form.submit()}>Login</button>
+                </div>
+            )
+        }
+    }
+}
+
+export default connect(
+    (state) => state.user
```

Comment:
Replace state.user with state.session here. connect() takes two optional arguments and returns a function which will pass the Redux state as props to whatever gets passed as its argument (aka Login component). If you just called connect()(Login), Login would be passed the complete redux state as this.props.state.

But we don't care about all the state in Login, we just want the relevant session state for login (whereas the signup component wants to look at the user state). So in order to not pass the whole state, we can pass a function known as mapStateToProps as the first arg to connect. It basically takes the whole redux state and you have to return what part of the state you actually want passed as props. In this case, we only want to look at state.session. Look at index.js in reducers to see what part of the state is mapped to what reducer. 

Just for completeness, the second optional argument to connect is called mapDispatchToProps. It takes the dispatch function, and you need to return all functions that you are going to use the dispatch function in. For example,  I use dispatch with actions.login(email, password), and I could have written mapDispatchToProps like so: (dispatch) => {login: (email, password) => dispatch(actions.login(email, password))} (and then used login function in onSubmit) but I only had one guy using dispatch so that would have been wordier. If you don't supply the second argument to connect, it automatically passes dispatch as props and that is why I was able to use this.props.dispatch...